enzyme/Enzyme/AdjointGenerator.h

```diff
@@ -1346,6 +1346,8 @@ class AdjointGenerator
   }
 
   Value *lookup(Value *val, IRBuilder<> &Builder) {
```

Comment:
I'd move this assertion within GradientUtils

Reply:
done c864b891711b8aa4953e97f347ca2176364251ef

Reply:
I might put this in a separate PR, since there is a lot of breakage with existing forward mode...

Reply:
#306